spec/presenters/tree_builder_alert_profile_obj_spec.rb

```diff
@@ -0,0 +1,126 @@
+describe TreeBuilderAlertProfileObj do
+  
+  before do
+   role = MiqUserRole.find_by(:name => "EvmRole-operator")
+   group = FactoryGirl.create(:miq_group, :miq_user_role => role, :description => "Tags Group")
+   login_as FactoryGirl.create(:user, :userid => 'tags_wilma', :miq_groups => [group])
+  end
+  
+  let(:tag1a) { FactoryGirl.create(:classification, :name => 'tag1a') }
+  let(:tag2a) { FactoryGirl.create(:classification, :name => 'tag2a') }
+  let(:tag3a) { FactoryGirl.create(:classification, :name => 'tag3a') }
+  let(:folder1a) do
+    f1 = FactoryGirl.create(:classification, :name => 'folder1a', :show => true)
+    f1.entries.push(tag1a)
+    f1.entries.push(tag2a)
+    f1.entries.push(tag3a)
+    f1
+  end
+  let!(:tag1b) { FactoryGirl.create(:tenant, :name => 'tag1b') }
+  let!(:tag2b) { FactoryGirl.create(:tenant, :name => 'tag2b') }
+  let!(:tag3b) { FactoryGirl.create(:tenant, :name => 'tag3b') }
+
+  context 'classification tree' do
+    subject do
+      described_class.new(:alert_profile_obj_tree, :alert_profile_obj,
+                          {},
+                          true,
+                          :assign_to => 'storage-tags',
+                          :cat       => folder1a.id,
+                          :objects   => [tag1a.id, tag2a.id])
+    end
+
+    describe '#tree_init_options' do
+      it 'sets init options correctly' do
+        expect(subject.send(:tree_init_options, :alert_profile_obj)).to eq(:expand => true)
+      end
+    end
+
+    describe '#set_locals_for_render' do
+      it 'sets locals for render correctly' do
+        locals = subject.send(:set_locals_for_render)
+        expect(locals[:check_url]).to eq("/miq_policy/alert_profile_assign_changed/")
+        expect(locals[:oncheck]).to eq("miqOnCheckHandler")
+        expect(locals[:checkboxes]).to be_truthy
+        expect(locals[:cfmeNoClick]).to be_truthy
+        expect(locals[:onclick]).to be_falsey
+      end
+    end
+
+    describe '#override' do
```

Comment:
Why? This is local variable, I need it only in the context of the function. They are not supposed to share anything between functions.